syntax = "proto3";
package sqlpkgs;




import "database_delivery_sdk/model/database_delivery/sql_package_version.proto";
import "database_delivery_sdk/model/database_delivery/app.proto";
import "database_delivery_sdk/model/database_delivery/dbservice.proto";

/*
CreateSQLPackage请求
*/
message CreateSQLPackageRequest {
message CreateSqlpkg {
    /*
    关联的应用 Id
    */
    string appId = 1 ;
    /*
    关联的数据库服务 Id
    */
    string dbServiceId = 2 ;
    /*
    名称
    */
    string name = 3 ;
    /*
    描述
    */
    string memo = 4 ;
}
        
    
    /*
    创建SQL包body
    */
    CreateSqlpkg createSqlpkg = 1 ;
}

/*
CreateSQLPackage返回
*/
message CreateSQLPackageResponse {
    /*
    SQL包下的版本列表
    */
    repeated database_delivery.SQLPackageVersion versionList = 1 ;
    /*
    SQL包关联的应用列表
    */
    repeated database_delivery.Application APP = 2 ;
    /*
    SQL包关联的数据库服务列表
    */
    repeated database_delivery.DBService DBSERVICE = 3 ;
    /*
    SQL包ID
    */
    string id = 4 ;
    /*
    名称
    */
    string name = 5 ;
    /*
    描述
    */
    string memo = 6 ;
    /*
    创建者
    */
    string creator = 7 ;
    /*
    ctime
    */
    int64 ctime = 8 ;
    /*
    mtime
    */
    int64 mtime = 9 ;
    /*
    文件仓库包ID
    */
    string repoPackageId = 10 ;
}

/*
CreateSQLPackageApi返回
*/
message CreateSQLPackageResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    sqlpkgs.CreateSQLPackageResponse data = 4 ;
}

